lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -173,31 +190,74 @@ def setup(self):
 
     def start(self):
 
-        self.data = read_data_dl2_to_QTable(str(self.input_dl2))
-        self.effective_time, self.elapsed_time = get_effective_time(self.data)
-        self.run_number = run_info_from_filename(self.input_dl2)[1]
+        if not self.source_dep:
+            self.data = read_data_dl2_to_QTable(str(self.input_dl2))
```

Comment:
If `read_data_dl2_to_QTable` does not support giving a `Path`, we should fix that function, not convert to string here.

Reply:
the `read_data_dl2_to_QTable` and `read_mc_dl2_to_QTable` uses `pandas.read_hdf` to read the files. It supports string or `os.PathLike` arguments for filename. So, either we convert all arguments to string within this function, or make it `os.PathLike`, by `pathlib.PurePath` right?

Reply:
The only reason to add `str()` here is that `read_data_dl2_to_QTable` does not support passing a `Path`. If it does support that, just remove `str`. If it doesn't support it, make it support it.

Reply:
it seems only `get_dataset_keys` , which is used to check if source-dependent parameters are included or not, does not support `Path` in `read_data_dl2_to_QTable` and `read_mc_dl2_to_QTable`. I added `str()` for `get_dataset_keys` in these functions

Reply:
perfect, thanks!

Reply:
Wait, I am confused, `get_dataset_keys` should already support `Path`.

Current master:

```
In [1]: from lstchain.io.io import get_dataset_keys

In [2]: from pathlib import Path

In [3]: path = Path('../data/gamma_20deg_0deg_run101___cta-prod5-paranal_desert-2147m-Paranal-dark_merged.DL1.h5')

In [4]: get_dataset_keys(path)
Out[4]: 
['/simulation/service/shower_distribution',
 '/simulation/event/subarray/shower',
 '/simulation/event/telescope/images/tel_001',
 '/simulation/event/telescope/images/tel_002',
 '/simulation/event/telescope/images/tel_003',
 '/simulation/event/telescope/images/tel_004',
```

What is the error you are getting?

Reply:
true... it seems I got some errors by what I introduced to check step by step...
so no need to use str anywhere!